web/src/components/NextPreviousButtons.test.js

```diff
@@ -0,0 +1,148 @@
+import { shallow, mount } from 'enzyme';
+import React from 'react';
+import { t } from '../i18n';
+
+import {
+  plain as NextPreviousButtons,
+  mapStateToProps,
+  mapDispatchToProps
+} from './NextPreviousButtons';
+
+import { jumpTo } from '../actions/app';
+
+let mockPush;
+
+jest.mock('react-router-dom', () => {
+  mockPush = jest.fn();
+  return {
+    useHistory: jest.fn().mockReturnValue({ push: mockPush }),
+    useRouteMatch: jest.fn().mockReturnValue({ path: '---path---' })
+  };
+});
+
+global.scrollTo = jest.fn();
+const pageNav = jest.fn();
+
+const context = { 
+  getLinks: jest.fn(),
+  getPreviousNextLinks: jest.fn().mockReturnValue([
+    {
+      id: 'apd-state-profile',
+      label: t('apd.stateProfile.title'),
+      onClick: pageNav('apd-state-profile-office', 'state-profile'),
+      children: [
+        {
+          id: 'apd-state-profile-office',
+          label: t('apd.stateProfile.directorAndAddress.title')
+        },
+        {
+          id: 'apd-state-profile-key-personnel',
+          label: t('apd.stateProfile.keyPersonnel.title')
+        }
+      ]
+    },
+    true,
+    {
+      id: 'apd-summary',
+      label: t('apd.title'),
+      onClick: pageNav('apd-summary', 'program-summary')
+    },
+    true
+  ])
+};
+
+const e = {
+  stopPropagation: jest.fn(),
+  preventDefault: jest.fn()
+};
+
+describe('NextPreviousButtons component', () => {
+  const props = {
+    activities: [
+      { anchor: '#key1', key: 'key 1234' },
+      { anchor: '#key2', key: 'key 4321' }
+    ],
+    activeSection: 'some section',
+    jumpTo: jest.fn(),
+    place: { id: 'place id', name: 'place name' },
+    context: context
+  };
+
+  test('renders correctly', () => {
+    const component = shallow(<NextPreviousButtons {...props} />);
+    expect(component).toMatchSnapshot();
+    expect(context.getLinks).toHaveBeenCalled();
+    expect(context.getPreviousNextLinks).toHaveBeenCalled();
+  });
+
+  it('handles previous button clicks', () => {
+    const component = shallow(<NextPreviousButtons {...props} />);
+    expect(component).toMatchSnapshot();
+
+    component.find('Button')
+      .at(0)
+      .simulate("click", e);
+
+    expect(pageNav).toHaveBeenCalledWith('apd-state-profile-office', 'state-profile');
+/*
+    // cancels the click
+    expect(e.stopPropagation).toHaveBeenCalled();
+    expect(e.preventDefault).toHaveBeenCalled();
+
+    // updates nav state, navigates, and scrolls up
+    expect(props.jumpAction).toHaveBeenCalledWith('activity-a1-overview');
+    expect(mockPush).toHaveBeenCalledWith('/apd/activity/0');
+    expect(window.scrollTo).toHaveBeenCalledWith(0, 0);
+*/
```

Comment:
The `NextPreviousButtons` internal `pageNav` and `anchorNav` methods aren't called directly by the component. Rather, it passes them into the `context.getLinks` method, which wraps them up and passes them back. So with the tests as-is, because you're mocking `context.getLinks`, you're able to test that the wrappers are called, but you're not getting into the internal methods.

To do that, you'll need to pull the internal methods out from the calls to the `context.getLinks` mock. Something like, e.g.:

```js
const internalPageNav = context.getLinks.mock[0][0];
const internalAnchorNav = context.getLinks.mock[0][1];
```

Then you can test the component's internal methods directly:

```js
const e = {
  stopPropagation: jest.fn(),
  preventDefault: jest.fn()
};

internalPageNav('id', 'route')(e);

expect(e.stopPropagation).toHaveBeenCalled();
expect(e.preventDefault).toHaveBeenCalled();

 // updates nav state, navigates, and scrolls up
 expect(props.jumpAction).toHaveBeenCalledWith('id');
 expect(mockPush).toHaveBeenCalledWith('/apd/route');
 expect(window.scrollTo).toHaveBeenCalledWith(0, 0);

internalAnchorNav('id');

expect(props.jumpAction).toHaveBeenCalledWith('id');
```

Reply:
thanks @mgwalker , I appreciate your taking a look!